app/src/main/java/com/example/fundatecheroes/character/view/NewCharacterActivity.kt

```diff
@@ -33,10 +37,9 @@ class NewCharacterActivity : AppCompatActivity(), OnItemSelectedListener {
 
         viewModel.viewState.observe(this) { state ->
             when (state) {
-                is
-                ViewState.ShowErrorNull -> toastCamposNull()
-                ViewState.ShowErrorUrl -> toastUrlInvalida()
-                ViewState.ShowErrorDate -> toastDataInvalida()
+                is ViewState.ShowErrorNull -> toastCamposNull()
+                is ViewState.ShowErrorHeroiVilao -> toastEscolhaHeroiVilao()
+                is ViewState.Loading -> loading()
                 is ViewState.ShowSuccess -> salvar()
```

Comment:
Como é apenas uma função que está sendo chamada, poderia ter chamado direto o `finish()` aqui
```suggestion
                is ViewState.ShowSuccess -> finish()
```